tests/test_authentication.py

```diff
@@ -66,3 +75,225 @@ async def test_authentication(fastapi_app, client, monkeypatch, httpx_mock: HTTP
+        assert (
+            client.request(endpoint[2], endpoint[1], headers={APIKEY_HEADER: VALID_APIKEY}).status_code
+            in expected_response
+        )
+
+    # With a valid api key in headers, set the download role
+    # Clear the cache
+    ttl_cache.clear()
+    httpx_mock.add_response(
+        url=RSPY_UAC_CHECK_URL,
+        match_headers={APIKEY_HEADER: VALID_APIKEY},
+        status_code=HTTP_200_OK,
+        json={"iam_roles": ["rs_adgs_download"], "config": {}},
+    )
+
+    for endpoint in endpoints:
+        if endpoint[0] == read_type:
+            expected_response = [HTTP_401_UNAUTHORIZED]
+        else:
+            # this means the authorization shall pass
+            expected_response = [HTTP_503_SERVICE_UNAVAILABLE, HTTP_404_NOT_FOUND]
+
+        assert (
+            client.request(endpoint[2], endpoint[1], headers={APIKEY_HEADER: VALID_APIKEY}).status_code
+            in expected_response
+        )
+
+    # With a valid api key in headers, set both of the download and read roles
+    # Clear the cache
+    ttl_cache.clear()
+    httpx_mock.add_response(
+        url=RSPY_UAC_CHECK_URL,
+        match_headers={APIKEY_HEADER: VALID_APIKEY},
+        status_code=HTTP_200_OK,
+        json={"iam_roles": ["rs_adgs_download", "rs_adgs_read"], "config": {}},
+    )
+
+    for endpoint in endpoints:
+        # this means the authorization shall pass
+        assert client.request(endpoint[2], endpoint[1], headers={APIKEY_HEADER: VALID_APIKEY}).status_code in [
+            HTTP_503_SERVICE_UNAVAILABLE,
+            HTTP_404_NOT_FOUND,
+        ]
+
+
+cadip_station_identifiers = ["ins", "mps", "mti", "nsg", "sgs", "cadip"]
+
+
+@pytest.mark.parametrize(
+    "fastapi_app",
+    [Envs({RSPY_LOCAL_MODE: False})],
+    ids=["authorization_adgs"],
+    indirect=["fastapi_app"],
+)
+def test_apikey_validator_cadip(
+    fastapi_app,
+    client,
+    monkeypatch,
+    httpx_mock: HTTPXMock,
+):  # pylint: disable=too-many-locals
+    """
+    Test that the http endpoints for the adgs station are protected and return 401 if not authorized.
+    Set RSPY_LOCAL_MODE to False before running the fastapi app.
+    """
+
+    # Mock the uac manager url
+    monkeypatch.setenv("RSPY_UAC_CHECK_URL", RSPY_UAC_CHECK_URL)
+
+    read_type = "read"
+    download_type = "download"
+    read_request_params = {"datetime": "2014-01-01T12:00:00Z/2023-02-02T23:59:59Z"}
+    download_request_params = {"name": "TEST_FILE.raw"}
+    set_access_type_read = []
+    set_access_type_download = []
+    for identifier in cadip_station_identifiers:
+        set_access_type_read.append(f"rs_{STATIONS_AUTH_lut[identifier]}_read")
+        set_access_type_download.append(f"rs_{STATIONS_AUTH_lut[identifier]}_download")
+    endpoints = []
+    # Gather the endpoints
+    # For each api endpoint (except the technical endpoints)
+    for route in fastapi_app.router.routes:
```

Comment:
Use a filter to remove bad routes.

`    filtered_routes = filter(
        lambda route: (isinstance(route, APIRoute) and route.path not in ("/", "/health") and "cadip" in route.path),
        fastapi_app.router.routes,
    )
`